### ISSUE TITLE
Get Exception AmqpEventHubConsumer.IdleTimerExpired: Idle timeout

### ISSUE THREAD

**Issue by tilumi** [Azure/azure-event-hubs-java#294]
I use azure-event-hub-spark 2.3.0 to consume data.
When consuming data with larger generation rate, I got a exception:
```
2018-03-20 08:45:53 INFO  ClientConnectionPool:54 - Client returned. EventHub name: wasclmonitors. Total clients: 2. Available clients: 1
2018-03-20 08:45:53 ERROR Executor:91 - Exception in task 11.0 in stage 17852.0 (TID 68882)
com.microsoft.azure.eventhubs.EventHubException: com.microsoft.azure.eventhubs.impl.AmqpException: The link 'G1:6144777:ca09dd_485d_G1_1521533703080' is force detached by the broker due to errors occurred in consumer(link2058397). Detach origin: AmqpEventHubConsumer.IdleTimerExpired: Idle timeout: 00:30:00. TrackingId:beda910f000103d4001f689d5ab0c307_G1_B0, SystemTracker:wascl-apc:eventhub:wasclmonitors~12287|sparktestcg, Timestamp:3/20/2018 8:45:53 AM, errorContext[NS: wascl-apc.servicebus.windows.net, PATH: wasclmonitors/ConsumerGroups/sparktestcg/Partitions/11, REFERENCE_ID: ca09dd_485d_G1_1521533703080, PREFETCH_COUNT: 1926, LINK_CREDIT: 0, PREFETCH_Q_LEN: 0]
	at com.microsoft.azure.eventhubs.impl.ExceptionUtil.toException(ExceptionUtil.java:54)
	at com.microsoft.azure.eventhubs.impl.MessageReceiver.onClose(MessageReceiver.java:615)
	at com.microsoft.azure.eventhubs.impl.BaseLinkHandler.processOnClose(BaseLinkHandler.java:50)
	at com.microsoft.azure.eventhubs.impl.BaseLinkHandler.handleRemoteLinkClosed(BaseLinkHandler.java:70)
	at com.microsoft.azure.eventhubs.impl.BaseLinkHandler.onLinkRemoteClose(BaseLinkHandler.java:36)
	at org.apache.qpid.proton.engine.BaseHandler.handle(BaseHandler.java:176)
	at org.apache.qpid.proton.engine.impl.EventImpl.dispatch(EventImpl.java:108)
	at org.apache.qpid.proton.reactor.impl.ReactorImpl.dispatch(ReactorImpl.java:324)
	at org.apache.qpid.proton.reactor.impl.ReactorImpl.process(ReactorImpl.java:291)
	at com.microsoft.azure.eventhubs.impl.MessagingFactory$RunReactor.run(MessagingFactory.java:445)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.microsoft.azure.eventhubs.impl.AmqpException: The link 'G1:6144777:ca09dd_485d_G1_1521533703080' is force detached by the broker due to errors occurred in consumer(link2058397). Detach origin: AmqpEventHubConsumer.IdleTimerExpired: Idle timeout: 00:30:00. TrackingId:beda910f000103d4001f689d5ab0c307_G1_B0, SystemTracker:wascl-apc:eventhub:wasclmonitors~12287|sparktestcg, Timestamp:3/20/2018 8:45:53 AM
	... 13 more
```

**Comment by SreeramGarlapati** [comment]
Hello @tilumi 
Does this repro consistently with large generation rate? what is the value ?
Thanks for reporting the issue!

**Comment by SreeramGarlapati** [comment]
I believe using spark adapter - `2.3.1` should alleviate this problem. Please try and let us know.

Thanks a lot @tilumi - for working with us towards resolution, much appreciated.

**Comment by SreeramGarlapati** [comment]
The repro detail you provided - `large generation rate` helped us to identify and fix this problem. We had a bug in `PartitionReceiver.setPrefetchCount()` API - which resulted in setting prefetchCount to Zero - which could render `PartitionReceiver` to idle situation. We fixed this issue in java client version `1.0.1`.
